nomos-services/data-availability/network/src/opinion_aggregator.rs

```diff
@@ -133,15 +139,20 @@ where
         }
     }
 
-    pub fn handle_session_change(&mut self, new_membership: Membership) -> Option<Opinions> {
-        // Generate opinions before clearing if we have both memberships
-        let opinions = if self.current_membership.is_some() && self.previous_membership.is_some() {
-            self.generate_opinions()
-        } else {
-            None
-        };
-
+    pub async fn handle_session_change(
+        &mut self,
+        new_membership: Membership,
+    ) -> Result<Option<Opinions>, DynError> {
```

Comment:
As this will always be called during the session change, I think the return type could be an `enum OpinionsResult`, which would contain all variants: `NoOpinions` `Opinions` `Error(DynError)`,